bigquery_storage/to_dataframe/main.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+
+def main(dataset_id="query_results"):
+    print("Creating clients.")
+    bqclient, bqstorageclient, project_id = create_clients()
+    print("\n\nReading a table:")
+    table_to_dataframe(bqclient, bqstorageclient)
+    print("\n\nReading query results:")
+    query_to_dataframe(bqclient, bqstorageclient, dataset_id)
+    print("\n\nReading a table, using the BQ Storage API directly:")
+    session_to_dataframe(bqstorageclient, project_id)
+
+
+def create_clients():
+    # [START bigquerystorage_pandas_create_client]
+    import google.auth
+    from google.cloud import bigquery
+    from google.cloud import bigquery_storage_v1beta1
+
+    # Explicitly create a credentials object. This allows you to use the same
+    # credentials for both the BigQuery and BigQuery Storage clients, avoiding
+    # unnecessary API calls to fetch duplicate authentication tokens.
+    credentials, project_id = google.auth.default(
+        scopes=['https://www.googleapis.com/auth/cloud-platform']
+    )
+
+    # Make clients.
+    bqclient = bigquery.Client(credentials=credentials, project=project_id)
+    bqstorageclient = bigquery_storage_v1beta1.BigQueryStorageClient(
+        credentials=credentials)
+    # [END bigquerystorage_pandas_create_client]
+    return bqclient, bqstorageclient, project_id
+
+
+def table_to_dataframe(bqclient, bqstorageclient):
+    from google.cloud import bigquery
+
+    # [START bigquerystorage_pandas_read_table]
+    # Download a table.
+    table = bigquery.TableReference.from_string(
+        "bigquery-public-data.utility_us.country_code_iso"
+    )
+    rows = bqclient.list_rows(
+        table,
+        selected_fields=[
+            bigquery.SchemaField("country_name", "STRING"),
+            bigquery.SchemaField("fips_code", "STRING"),
+        ]
+    )
+    dataframe = rows.to_dataframe(bqstorageclient)
+    print(dataframe.head())
+    # [END bigquerystorage_pandas_read_table]
+
+
+def query_to_dataframe(bqclient, bqstorageclient, dataset_id):
+    from google.cloud import bigquery
+
+    # [START bigquerystorage_pandas_read_query_results]
+    import uuid
+
+    # Due to a known issue in the BigQuery Storage API (TODO: link to
```

Comment:
Should we consider simply running a large query that emits enough data to avoid the inline?  Pros:  better demonstrates the perf of the new API, and avoids us having to revisit the sample.  Cons:  test time overhead and potential pitfalls for people kicking tires with small results.   Part of this is dependent on how the team will be maintaining their KI list.

Reply:
> Cons: test time overhead.

Long test time is my main reason for avoiding queries that return big results. I guess it's not so bad since this repo can test the different directories independently.

> Cons: potential pitfalls for people kicking tires with small results.

The current failure case is rather bad: it returns a successful response, but gives you an empty result set. That's a pretty big pitfall, because catching it requires noticing that you didn't get the data you thought you were going to get.

Reply:
Works for me.  Let's confirm how the KIs will be maintained so you can link them.